### DISCUSSION TITLE
Find longest path in a graph with cycles

### DISCUSSION THREAD

**Discussion by RobinFrcd** [networkx/networkx#5989] [Q&A]
Hi,
I'm working with a graph that has cycles. I'd like to know if there's a function like [`dag_longest_path` ](https://networkx.org/documentation/stable/reference/algorithms/generated/networkx.algorithms.dag.dag_longest_path.html) but when dealing with cycles ?

My use case is, given a directed graph with cycles and a node, I'd like to find the path that goes to as far as possible from the node with a given maximum depth. The distance here would be the euclidean distance between the points, not the costs of the edges.

Thanks !

**Comment by dschult** [comment]
We don't have functions that view path distance as euclidean distance --
that is an embedding problem (how is the network embedded in 3-space). So
it's not really a network problem. Numpy probably has functions that would
compute the euclidean distance between all points and then you'd find the
max from that.

Notice that the network structure is irrelevant for finding the max
euclidean distance so long as the network is strongly connected.

**Reply by RobinFrcd** [reply]
Thank you for your answer. So even if I set the edge cost to be the euclidean distance between two points, there's no way do to it with an implemented solution in networkx ?

Thanks

**Reply by dschult** [reply]
You can make the edge cost to be the euclidean distance, but then a 2-edge path does not have the distance of the sum of the edges. Right?  node A and B could be 1 meter apart with a 2-edge path through node C that is 500 meters away. The path length would show as 1000 meters but you want the path length to be 1m.  But maybe I am misunderstanding.

**Reply by RobinFrcd** [reply]
Well, in my particular case, that's what I need because I'm working with train journeys. So I'd indeed just need to sum the distance between nodes.

**Reply by dschult** [reply]
Then what do you mean by "longest path" with cycles?  Going around the cycles more often will just add more and more length and there would be no longest path.

**Reply by RobinFrcd** [reply]
Well, I guess I just forgot a constraint in my problem, indeed. :sweat_smile: I need to maximize the euclidean distance between nodes while still having the sum of distances between intermediate nodes stored somewhere.

I guess I'm indeed going to need numpy here, thanks !
